traffic_server/datadog_checks/traffic_server/check.py

```diff
@@ -0,0 +1,113 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import Any, Dict, List, Optional
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+
+from .metrics import HOSTNAME_METRIC_NAMES, SHORT_VERSION_METRIC_NAMES, VERSION_BUILD_NUMBER_METRIC_NAME, build_metric
+
+NULL_VALUE = "(null)"
+SERVICE_CHECK_CONNECT = "can_connect"
+
+
+class TrafficServerCheck(AgentCheck):
+
+    __NAMESPACE__ = 'traffic_server'
+
+    def __init__(self, name, init_config, instances):
+        super(TrafficServerCheck, self).__init__(name, init_config, instances)
+
+        self.traffic_server_url = self.instance.get("traffic_server_url")
+        self.tags = self.instance.get("tags", [])
+
+        if self.traffic_server_url is None:
+            raise ConfigurationError('Must specify a traffic_server_url')
+
+    def check(self, _):
+        # type: (Any) -> None
+        hostname_tag = []
+
+        try:
+            response = self.http.get(self.traffic_server_url)
+            response.raise_for_status()
+            response_json = response.json()
+            global_metrics = response_json.get("global")
+            if global_metrics is None:
+                self.log.warning(
+                    "Could not parse traffic server metrics payload, skipping metric and version collection"
+                )
+                self.service_check(SERVICE_CHECK_CONNECT, AgentCheck.WARNING, tags=self.tags)
+                return
+
+            hostname_tag = self.get_hostname_tag(global_metrics)
+            self.collect_metrics(global_metrics, hostname_tag)
+            self.collect_version(global_metrics)
+
+        except Exception as e:
+            self.service_check(
+                SERVICE_CHECK_CONNECT,
+                AgentCheck.CRITICAL,
+                tags=self.tags + hostname_tag,
+                message="Request failed: {}, {}".format(self.traffic_server_url, e),
+            )
+            raise
+
+        self.service_check(SERVICE_CHECK_CONNECT, AgentCheck.OK, tags=self.tags)
+
+    def get_hostname_tag(self, global_metrics):
+        # type: (Dict[str, Any]) -> List[str]
+        for hostname_metric in HOSTNAME_METRIC_NAMES:
+            hostname_value = global_metrics.get(hostname_metric)
+            if hostname_value is not None and hostname_value != NULL_VALUE:
+                return ['traffic_server_host:{}'.format(hostname_value)]
```

Comment:
If both `proxy.node.hostname_FQ` and `proxy.node.hostname` are valid and have different values, does one of them have higher "priority" or is it just whichever hostname is first?

Reply:
`proxy.node.hostname_FQ` has a higher "priority" since it's first in the `HOSTNAME_METRIC_NAMES` list, and the function returns when it finds the first hostname. https://github.com/DataDog/integrations-core/pull/11471/files#diff-6629e3ff1299680c342a4bddbfa7311acf0427d50f908cb00e95fcf55d00bd99R63

Reply:
Is there any chance that the discovered hostname would change between check runs? I.e. does the every payload consistently return the same hostname whether it's defined as `proxy.node.hostname_FQ` or `proxy.node.hostname`? Just wondering if that could cause tag flapping

Reply:
From my testing if `proxy.node.hostname_FQ` is present, `proxy.node.hostname` is also present, unlike the version metrics which sometimes have null values. It does seem cleaner to just have  `proxy.node.hostname_FQ` as the designated hostname, and just not send the tag if it is not available. What do you think?